h/features/models.py

```diff
@@ -105,3 +107,32 @@ def remove_old_flags(cls):
 
     def __repr__(self):
         return '<Feature {f.name} everyone={f.everyone}>'.format(f=self)
+
+
+class FeatureCohort(Base, mixins.Timestamps):
+    __tablename__ = 'featurecohort'
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    name = sa.Column(sa.UnicodeText(), nullable=False, index=True)
+
+    # Cohort membership
+    members = sa.orm.relationship(
+        'User', secondary='featurecohort_user', backref=sa.orm.backref(
+            'cohorts', order_by='FeatureCohort.name'))
+
+    def __init__(self, name):
+        self.name = name
+
+
+USER_COHORT_TABLE = sa.Table(
+    'featurecohort_user', Base.metadata,
+    sa.Column('id', sa.Integer, nullable=False),
+    sa.Column('cohort_id',
+              sa.Integer,
+              sa.ForeignKey('featurecohort.id'),
+              nullable=False),
+    sa.Column('user_id',
+              sa.Integer,
+              sa.ForeignKey('user.id'),
+              nullable=False),
```

Comment:
Add a `sa.UniqueConstraint` here that guarantees only one row can exist for a given (user, cohort) pair.